crates/e2e/tests/e2e/flashloans.rs

```diff
@@ -110,30 +111,25 @@ async fn forked_mainnet_repay_debt_with_collateral_of_safe(web3: Web3) {
     // could be replaced with `permit` pre-hook or `repayWithPermit()` for
     // borrowed tokens that support `permit`
     trader
-        .exec_call(
-            onchain
-                .contracts()
-                .weth
-                .approve(aave_pool.address(), to_wei(1)),
-        )
+        .exec_call(weth.approve(aave_pool.address(), to_wei(1)))
         .await;
 
+    // pay some extra for the flashloan fee
+    let fee_bps = aave_pool.flashloan_premium_total().call().await.unwrap();
+    let flashloan_fee = (flashloan_amount * U256::from(fee_bps)).ceil_div(&10_000.into());
+
     let current_safe_nonce = trader.nonce().await;
 
     // Build appdata that does:
     // 1. take out a 1 WETH flashloan for the trader (flashloan hint)
     // 2. repay the 1 WETH debt to unlock trader's collateral (1st pre-hook)
     // 3. withdraw the collateral it can be sold for WETH (2nd pre-hook)
+    // 4. repay the flashloan with the proceeds (1st post-hook)
```

Comment:
In the e2e test we can't easily assume 0 fee. For the test to work with the latest approach we'd still have to have this post-interaction **but only for repaying the fee** not the the underlying flashloaned amount. Repayment of the raw flashloan amount should be handled inside the driver (by injecting the post-interaction in the settlement call as discussed above).